From 3869a662d8cf6be32a6b83f7360848763d794490 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:13 -0400
Subject: [tools] perf/machine: Use snprintf instead of sprintf

Message-id: <1408462094-14194-145-git-send-email-jolsa@redhat.com>
Patchwork-id: 88175
O-Subject: [PATCH RHEL7.1 BZ1131394 144/325] perf machine: Use snprintf instead of sprintf
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit f4be904d2fa7c65e230ed4b1008ebdf5f4053ac3
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Sun Sep 22 13:22:09 2013 +0300

description
===========
To avoid buffer overruns.

[ Split from aa7fe3b ]
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/machine.c b/tools/perf/util/machine.c
index 6188d28..ddf917b 100644
--- a/tools/perf/util/machine.c
+++ b/tools/perf/util/machine.c
@@ -785,10 +785,10 @@ static int machine__create_modules(struct machine *machine)
  const char *modules;
  char path[PATH_MAX];
 
- if (machine__is_default_guest(machine))
+ if (machine__is_default_guest(machine)) {
   modules = symbol_conf.default_guest_modules;
- else {
-  sprintf(path, "%s/proc/modules", machine->root_dir);
+ } else {
+  snprintf(path, PATH_MAX, "%s/proc/modules", machine->root_dir);
   modules = path;
  }
 
-- 
1.7.1